mount.go

```diff
@@ -137,10 +138,49 @@ func doMount(args *argContainer) {
 	// Return memory that was allocated for scrypt (64M by default!) and other
 	// stuff that is no longer needed to the OS
 	debug.FreeOSMemory()
+	// Set up autounmount, if requested.
+	if args.autounmount > 0 && !args.reverse {
+		// Not being in reverse mode means we always have a forward file system.
+		fwdFs := fs.(*fusefrontend.FS)
+		go idleMonitor(args.autounmount, fwdFs, srv, args.mountpoint)
+	}
 	// Jump into server loop. Returns when it gets an umount request from the kernel.
 	srv.Serve()
 }
 
+// Based on the EncFS idle monitor:
+// https://github.com/vgough/encfs/blob/1974b417af189a41ffae4c6feb011d2a0498e437/encfs/main.cpp#L851
+const ActivityCheckIntervalSeconds = 10
+// idleMonitor is a function to be run as a thread that checks for
+// filesystem idleness and unmounts if we've been idle for long enough.
+func idleMonitor(idleTimeout int, fs *fusefrontend.FS, srv *fuse.Server, mountpoint string) {
+	timeoutCycles := 60 * idleTimeout / ActivityCheckIntervalSeconds
+	idleCount := 0
+	for true {
+		recentAccess := fs.AccessedSinceLastCheck
+		// Don't worry about flag being set here by another thread...
+		// on the scale of minutes, it might as well have happened
+		// just before we checked.
+		fs.AccessedSinceLastCheck = false
```

Comment:
Just to be safe, should we use CompareAndSwapUint32 from the atomic package here? This is literally just a compare-and-swap operation, after all.

Reply:
Yes!

Reply:
By the way, I think integer math would be good enough here - time.Duration values are in nanoseconds